.github/workflows/npm-cd.yml

```diff
@@ -24,6 +24,9 @@ jobs:
         steps:
           - name: Checkout
             uses: actions/checkout@v4
+            with:
+                fetch-depth: 0
+                ref: ${{ github.head_ref }}
```

Comment:
> The trick is that github.head_ref is only set when the workflow was triggered by a pull_request and it contains the value of the source branch of the PR. github.ref_name will than only be used if the workflow was not triggered by a pull_request and it also just contains the branch name.


https://stackoverflow.com/a/71158878
so it might fail on push.
Why do we need this change? isn't it the default behavior anyways?

Reply:
I'll try without the ref.
the depth is needed because sometime the cd on the self hosted dont take the last commits
I wanted to give the ref so it for sure will know which ref i need, but lets se if sufficient with depth only